src/hera/workflows/script.py

```diff
@@ -330,6 +446,13 @@ def _get_parameters_and_artifacts_from_callable(
     return parameters, artifacts
 
 
+def _check_hera_outputs_used(source: Callable) -> bool:
+    """Check if hera outputs are used. This is needed to know if we need to create a volume."""
```

Comment:
This is only checking if output annotations are used - can we rename? Current name sounds like whether the default `hera/outputs` directory is used. Also don't need to say what this function is used for as that can change over time. Also, might be better to put the experimental features check here rather than at the call site in the other comment

```suggestion
def _output_annotations_used(source: Callable) -> bool:
    """Check if hera output annotations are used."""
```